### PR TITLE
Refactor Delete Instance Wizard in Tentacle Manager

### PR DESCRIPTION
**Pull request by acodrington**
Move some :cheese: in the Tentacle Manager's "Delete Instance" wizard.

# Background
We want to add some automated tests to Tentacle Manager as part of the [Tentacle Manager on .NET](https://app.shortcut.com/octopusdeploy/epic/68113/tentacle-manager-on-windows-net-core) project.

We first need to make the Tentacle Manager project testable by automated tests (e.g. NUnit) by separating the UI logic from the business logic. This will allow the automated tests to exercise most of the code (i.e. everything _not_ in a UI component) without the pain of dealing with UI threads and Dispatchers etc.

[sc-70325]

# Results

There should be *zero visible changes* from an end-user's perspective. The intent is that the functionality is completely unchanged, but the code base is in a better shape for automated tests.

Summary of the main changes within:

#### Remove `ShellView` factory method from container
UI components must be created on the UI thread, which gets real messy when the container is involved. It's best to just not do it, and only use the container for resolving models and other business logic components.

#### `DeleteWizardModel` and `TentacleManagerModel` now derive from `ShellViewModel` 
Existing logic means that multiple models must be created when opening a new Tentacle Manager window - one for the `ShellView`, and one for the `ShellView`'s UI content.

`ShellViewModel` only contains properties that are common to all features being hosted in the `ShellView`, so it makes sense to simplify window creation by allowing `ShellView` to use the same view model as its content.

#### Remove `DeleteWizardLauncher`
This class' only responsibility was to create the models and views required for the "Delete Instance" wizard. Creating models will now be done via the container, and creating the views will be done manually within the UI logic, so the UI creation functionality has been moved to `TentacleManagerView` where it is used.

#### `TentacleManagerModel` creates the `DeleteWizardModel`
It should not be the responsibility of a UI component to know how to create models for a child UI component. `DeleteWizardModel` is now registered in the container, and `TentacleManagerModel` depends on a factory function to create them on demand i.e. whenever the Tentacle Manager view wants to open a new instance of the "Delete Instance" wizard.

#### `TentacleManagerView` depends on the container
While the end goal is to have views relatively ignorant of how to create models, iteratively implementing this means we need to let `TentacleManagerView` create/resolve components it probably shouldn't be... at least until we've converted all the views/components.

#### Add exclusion for local dev builds to the `IsEarlyAccessProgram` helper method
By removing some of the usages of the container to resolve `ShellViewModel`, a side effect was introduced. [This line, which has been removed,](https://github.com/OctopusDeploy/OctopusTentacle/compare/sast/tentacle-manager/delete-wizard?expand=1#diff-77d439cef0dfaaf630696e22466d1bc6b423d232a96943d3f0bdace00c5f73acL21) added a handler action for the container's OnActivating event when resolving a `ShellViewModel`. This action modified the resolved instance by disabling the EAP message. While it is highly unlikely we will be releasing any EAP versions of Tentacle Manager in the future, rather than delete this functionality it has been updated so that the EAP message gets disabled if the build is a local dev build i.e. Major version starts with 0, such as `0.0.0-local`.

# How to review this PR

Quality :heavy_check_mark:

As there are no automated tests, the only way to verify this PR is to manually test the Tentacle Manager. There's not much functionality in here, so testing the following should cover everything:
* Install new instance
* Reinstall existing instance
* Restart service
* Start service
* Stop service
* Update proxy settings
* Delete instance

# Pre-requisites

- [x] I have read [How we use GitHub Issues](https://github.com/OctopusDeploy/Issues/blob/main/docs/CONTRIBUTING.internal.md) for help deciding when and where it's appropriate to make an issue.
- [x] I have considered informing or consulting the right people, according to the [ownership map](https://whimsical.com/ownership-map-NzbiD4HJyvhC9jNJNfS6TG).
- [x] I have considered appropriate testing for my change.

### CODE REVIEW

File: source/Octopus.Manager.Tentacle/DeleteWizard/Views/DeleteWelcome.xaml [hunk 1]
```diff
@@ -30,7 +29,7 @@
 
                 <StackPanel Margin="10,0,10,0">
                     <TextBlock Style="{StaticResource TextBlock.Big}" TextWrapping="Wrap">
-                    <Run Text="{Binding Path=ApplicationName, StringFormat='Are you sure you want to delete your {0} instance?'}" />
+                    <Run Text="{Binding Mode=OneWay, Path=ApplicationName, StringFormat='Are you sure you want to delete your {0} instance?'}" />
```
Comment (LukeButters) [inline source/Octopus.Manager.Tentacle/DeleteWizard/Views/DeleteWelcome.xaml:32]: I have no idea what this means :)

**Comment by LukeButters** [comment]
When I was trying this out on a machine with many tentacles installed I got an error starting the service:
```
Object reference not set to an instance of an object.
System.NullReferenceException
   at Octopus.Manager.Tentacle.TentacleConfiguration.TentacleManager.TentacleManagerView.StartServiceClicked(Object sender, EventArgs e) in TentacleManagerView.xaml.cs:line 109
   at System.Windows.EventRoute.InvokeHandlersImpl(Object source, RoutedEventArgs args, Boolean reRaised)
   at System.Windows.UIElement.RaiseEventImpl(DependencyObject sender, RoutedEventArgs args)
   at System.Windows.Controls.Primitives.ButtonBase.OnClick()
   at System.Windows.Controls.Button.OnClick()
   at System.Windows.Controls.Primitives.ButtonBase.OnMouseLeftButtonUp(MouseButtonEventArgs e)
   at System.Windows.RoutedEventArgs.InvokeHandler(Delegate handler, Object target)
   at System.Windows.RoutedEventHandlerInfo.InvokeHandler(Object target, RoutedEventArgs routedEventArgs)
   at System.Windows.EventRoute.InvokeHandlersImpl(Object source, RoutedEventArgs args, Boolean reRaised)
   at System.Windows.UIElement.ReRaiseEventAs(DependencyObject sender, RoutedEventArgs args, RoutedEvent newEvent)
   at System.Windows.UIElement.OnMouseUpThunk(Object sender, MouseButtonEventArgs e)
   at System.Windows.RoutedEventArgs.InvokeHandler(Delegate handler, Object target)
   at System.Windows.RoutedEventHandlerInfo.InvokeHandler(Object target, RoutedEventArgs routedEventArgs)
   at System.Windows.EventRoute.InvokeHandlersImpl(Object source, RoutedEventArgs args, Boolean reRaised)
   at System.Windows.UIElement.RaiseEventImpl(DependencyObject sender, RoutedEventArgs args)
   at System.Windows.UIElement.RaiseTrustedEvent(RoutedEventArgs args)
   at System.Windows.Input.InputManager.ProcessStagingArea()
   at System.Windows.Input.InputManager.ProcessInput(InputEventArgs input)
   at System.Windows.Input.InputProviderSite.ReportInput(InputReport inputReport)
   at System.Windows.Interop.HwndMouseInputProvider.ReportInput(IntPtr hwnd, InputMode mode, Int32 timestamp, RawMouseActions actions, Int32 x, Int32 y, Int32 wheel)
   at System.Windows.Interop.HwndMouseInputProvider.FilterMessage(IntPtr hwnd, WindowMessage msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   at System.Windows.Interop.HwndSource.InputFilterMessage(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   at MS.Win32.HwndWrapper.WndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   at MS.Win32.HwndSubclass.DispatcherCallbackOperation(Object o)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)
```

**Review by LukeButters — APPROVED** [review]
LGTM although I couldn't get tentacle manager working well enough to test it out.

File: source/Octopus.Manager.Tentacle/DeleteWizard/Views/DeleteWelcome.xaml (hunk 1 above)
Comment (raynhamdev) [reply source/Octopus.Manager.Tentacle/DeleteWizard/Views/DeleteWelcome.xaml:32]: default Binding option is "TwoWay", which means changes in UI are synced with the code behind it and vice versa. For this view, we have some read-only properties in the code behind, which may confuse the default binding option. Hence, it is set to "OneWay".

**Review by raynhamdev — APPROVED** [review]
Changes LGTM.
Testing the branch locally and saw this error showing up after starting Tentacle Manager.
image

However, I confirmed this is also happening in `main` so it is not caused by the changes in this PR.

**Pull request merged**